File: JavaScript/프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    var answer = 0;
    const stack = [];
    for (const alpha of s){
        if (stack.length === 0) {
            stack.push(alpha);
            continue;
        }
        if (stack[stack.length - 1] === alpha) stack.pop();
        else {
            stack.push(alpha);
        }
    }
    if (stack.length) answer = 0;
    else answer = 1;
    return answer;
}

File: C++/프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    int answer = -1;
    stack<char> stk;
    
    for (int i = 0; i < s.size(); i++){
        char temp = s[i];
        if (stk.empty()){
            stk.push(temp);
        } else {
            if (stk.top() == temp) {
                stk.pop();
            } else {
                stk.push(temp);
            }
        }
    }
    if (stk.empty()) answer = 1;
    else answer = 0;
    return answer;
}